src/mds/Server.cc

```diff
@@ -2556,6 +2556,12 @@ void Server::handle_client_reply(const cref_t<MClientReply> &reply)
     if (dn) {
       dn->state_clear(CDentry::STATE_REINTEGRATING);
 
+      // move the state back to STATE_UNLINKING
+      if (dn->state_test(CDentry::STATE_UNLINK_REINTEGRATING)) {
+        dn->state_set(CDentry::STATE_UNLINKING);
+        dn->state_clear(CDentry::STATE_UNLINK_REINTEGRATING);
```

Comment:
shouldn't we first clear the state and then set?

Reply:
> shouldn't we first clear the state and then set?

This should be okay since the it always holding the global `mds_lock`.
But I will fix this anyway. 

Thanks @dparmar18